python/lsst/ap/association/association.py

```diff
@@ -129,13 +129,36 @@ def run(self,
 
         diaObjects = diaObjects.append(matchResult.new_dia_objects,
                                        sort=True)
+        # if len(diaObjects) > 0:
+        #     dups = diaObjects.iloc[[0, -1]]
+        #     diaObjects = diaObjects.append(dups, sort=True)
+        if diaObjects.index.has_duplicates:
+            self.log.warn(
+                "Duplicate DiaObjects created after association. This may "
+                "cause downstream pipeline issues. Dropping duplicated rows.")
+            # Drop duplicates via index and keep the first appearance.
+            diaObjects = diaObjects.groupby(diaObjects.index).first()
```

Comment:
Here, and throughout these changes where you use `groupby`: does this change the ordering of the diaObjects?

Reply:
It might, but since the dataFrames are indexed and continue to be so after this interaction it won't matter. I rarely use iloc and only do so after matching it to the proper location within the arrays.